### PR TITLE
icu4c 73.2 (merge from staging branch)

### PR DESCRIPTION
**Pull request by carlocab**
Opening this to track progress for `icu4c`, and to apply https://github.com/Homebrew/homebrew-core/labels/no%20long%20build%20conflict.

### CODE REVIEW

**Comment by carlocab** [comment]
These are all the formulae with linkage to `icu4c` on Linux: https://github.com/carlocab/workflow-test/actions/runs/5287851542#summary-14319124418

While it's technically possible for a formula to have linkage with `icu4c` only on macOS, that seems sufficiently unlikely to me that we should be good to merge this to `master` once the outstanding PRs are merged to the staging branch.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Great work here @carlocab!

**Pull request merged**
